### PR TITLE
Move library to .NET 5, and use a custom class for time and date.

### PR DESCRIPTION
**Pull request by danielalvsaaker**
- Now compatible with newest stable version of .NET
- Implemented the TADDateTime class for storing edge case dates.
- Updated/altered tests to reflect changes

### CODE REVIEW

File: TimeAndDate.Services/DataTypes/Time/TADDateTime.cs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        {
+            Year = date.Year;
+            Month = date.Month;
+            Day = date.Day;
+            Hour = date.Hour;
+            Minute = date.Minute;
+            Second = date.Second;
+        }
+
+        public DateTimeOffset ToStd()
```
Comment (ninjacato) [inline TimeAndDate.Services/DataTypes/Time/TADDateTime.cs]: I think you can use explicit operator conversion here
```
public static explicit operator DateTimeOffset(TADDateTime date)
{
        return new DateTimeOffset(date.Year, etc...)
}
 ```

So people can just cast it
```
var taddate = new TADDateTime(etc, etc);
var offset = (DateTimeOffset)taddate;
```

**Review by ninjacato — CHANGES_REQUESTED** [review]

**Review by ninjacato — APPROVED** [review]
